Kernel/TTY/VirtualConsole.cpp

```diff
@@ -134,12 +134,54 @@ UNMAP_AFTER_INIT void VirtualConsole::initialize()
 
     // Add the lines, so we also ensure they will be flushed now
     for (size_t row = 0; row < rows(); row++) {
-        m_lines.append({ true });
+        m_lines.append({ true, 0 });
     }
     clear();
     VERIFY(m_cells);
 }
 
+void VirtualConsole::refresh_after_resolution_change()
+{
+    auto old_rows_count = rows();
+    auto old_columns_count = columns();
+    set_size(GraphicsManagement::the().console()->max_column(), GraphicsManagement::the().console()->max_row());
+    m_console_impl.set_size(GraphicsManagement::the().console()->max_column(), GraphicsManagement::the().console()->max_row());
+
+    // Note: From now on, columns() and rows() are updated with the new settings.
+
+    auto size = GraphicsManagement::the().console()->max_column() * GraphicsManagement::the().console()->max_row() * sizeof(Cell) * 2;
+    auto new_cells = MM.allocate_kernel_region(page_round_up(size), "Virtual Console Cells", Region::Access::Read | Region::Access::Write, AllocationStrategy::AllocateNow);
+
+    if (rows() < old_rows_count) {
+        m_lines.shrink(rows());
+    } else {
+        for (size_t row = 0; row < (size_t)(rows() - old_rows_count); row++) {
+            m_lines.append({ true, 0 });
+        }
+    }
+
+    // FIXME: Potential loss of displayed data occur when resolution width shrinks.
```

Comment:
That's totally fine. If someone is changing the resolution, there's a 100% chance they will redraw the screen afterwards anyway :)